service-base/src/main/java/org/eclipse/hono/service/http/DefaultFailureHandler.java

```diff
@@ -75,6 +76,32 @@ public void handle(final RoutingContext ctx) {
         }
     }
 
+    private String extractMessage(final HttpServerResponse response) {
+        if ( response == null ) {
+            return MSG_UNKNOWN_ERROR;
```

Comment:
this code is unreachable, we will have run into an NPE in line 56 before we arrive here ...